### PR TITLE
#30 #162 delete course has module connection

### PR DESCRIPTION
**Pull request by BojanMirkovic**
# DeleteCourseHasModuleConnection

## Description
I have created Endpoint,Command + Handler, Repository method

## Changes Made
DeleteCourseCommand + Handler are updated 

## Related Ticket or Issue
#30 
#162 
#98 

## Checklist

- [x] Code adheres to the coding conventions
- [x] Unit tests are added for new code
- [x] Existing unit tests are updated if needed
- [x] Code follows the principles of Clean Architecture
- [x] Code formatting complies with the `.editorconfig` file
- [x] Pull request template is filled out

### CODE REVIEW

File: API-Layer/Controllers/CourseController.cs [hunk 1]
```diff
@@ -92,6 +93,27 @@ public async Task<IActionResult> DeleteCourse(string courseId)
                 return BadRequest(ex.Message);
             }
         }
+        [HttpDelete("DeleteCourseHasModuleConnection/{courseId} , {moduleId}")]
```
Comment (HubTicaBB) [inline API-Layer/Controllers/CourseController.cs:96]: Same as in [the other PR](https://github.com/InFiNet-Code-AB/Dojo-BE/pull/166#discussion_r1602695688), I'd suggest renaming the route to something more intuitive, e.g.
- `courses/{courseId}/remove-module/{moduleId}`
- `courses/{courseId}/module/{moduleId}/remove`

**Comment by HubTicaBB** [comment]
Looks good, otherwise! 👍

File: API-Layer/Controllers/CourseController.cs (hunk 1 above)
Comment (BojanMirkovic) [reply API-Layer/Controllers/CourseController.cs]: route name is changed as sugested

**Review by ElliotDahlin — APPROVED** [review]
looks good

**Pull request merged**